CMakeLists.txt

```diff
@@ -14,6 +14,19 @@ include(sources.cmake)
 enable_language(C)
 enable_language(CXX)
 
+# Fail CMake build when some vulnerable gcc versions are used.
+set(VUL_GCC_VERSIONS "")
+# These versions are disabled due to a bug reported in memcmp, which is fundamental and crucial for crypto.
+# https://gcc.gnu.org/bugzilla/show_bug.cgi?id=95189
+list(APPEND VUL_GCC_VERSIONS "9.2" "9.3" "10.1")
+if(CMAKE_COMPILER_IS_GNUCXX)
+  foreach(vul_gcc_version ${VUL_GCC_VERSIONS})
+    if("${vul_gcc_version}" VERSION_EQUAL CMAKE_C_COMPILER_VERSION)
+      message(FATAL_ERROR "GCC ${CMAKE_C_COMPILER_VERSION} is not supported.")
```

Comment:
It maybe hard to add specific reason because I think other gcc versions maybe disabled here due to diff bugs.

This is why below style/pattern is  created.
```
# explanation of why disable the gcc
list(APPEND VUL_GCC_VERSIONS "new version")
```

```
# These versions are disabled due to a bug reported in memcmp, which is fundamental and crucial for crypto.
# https://gcc.gnu.org/bugzilla/show_bug.cgi?id=95189
list(APPEND VUL_GCC_VERSIONS "9.2" "9.3" "10.1")
if(CMAKE_COMPILER_IS_GNUCXX)
  foreach(vul_gcc_version ${VUL_GCC_VERSIONS})
    if("${vul_gcc_version}" VERSION_EQUAL CMAKE_C_COMPILER_VERSION)
      message(FATAL_ERROR "GCC ${CMAKE_C_COMPILER_VERSION} is not supported.")
    endif()
  endforeach()
endif()
```

Reply:
Also, couldn't this be simplified as just:

```
if(CMAKE_C_COMPILER_VERSION IN ${VUL_GCC_VERSIONS})
  message(FATAL_ERROR "GCC ${CMAKE_COMPILER_VERSION} is not supported.")
endif()
```

Reply:
Nevermind, apparently `IN_LIST` was not added until version 3.3

Reply:
I meant more in the form of a more explanatory error. But having short error message is maybe also a good idea. I'll leave it to you to decide what you feel is best.